src/transformers/models/falcon_mamba/modeling_falcon_mamba.py

```diff
@@ -745,11 +773,17 @@ def prepare_inputs_for_generation(
         else:
             model_inputs = {"input_ids": input_ids.contiguous()}
 
+        # In case cache is not used, manually add a new column in the attention mask
+        if not use_cache and attention_mask is not None and input_ids.shape != attention_mask.shape:
+            padd_length = input_ids.shape[-1] - attention_mask.shape[-1]
+            attention_mask = torch.cat([attention_mask, torch.ones_like(input_ids[:, :padd_length])], dim=-1)
```

Comment:
small typo!
```suggestion
        # In case cache is not used, manually add a new column in the attention mask
        if not use_cache and attention_mask is not None and input_ids.shape != attention_mask.shape:
            pad_length = input_ids.shape[-1] - attention_mask.shape[-1]
            attention_mask = torch.cat([attention_mask, torch.ones_like(input_ids[:, :pad_length])], dim=-1)
```